testingutils/src/main/java/com/github/awsjavakit/testingutils/networking/CustomRequest.java

```diff
@@ -0,0 +1,182 @@
+package com.github.awsjavakit.testingutils.networking;
+
+import com.github.awsjavakit.misc.JacocoGenerated;
+import com.github.tomakehurst.wiremock.http.ContentTypeHeader;
+import com.github.tomakehurst.wiremock.http.Cookie;
+import com.github.tomakehurst.wiremock.http.FormParameter;
+import com.github.tomakehurst.wiremock.http.HttpHeader;
+import com.github.tomakehurst.wiremock.http.HttpHeaders;
+import com.github.tomakehurst.wiremock.http.QueryParameter;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.RequestMethod;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class CustomRequest implements Request {
+
+  private final Request request;
+  private final Map<String, FormParameter> formParametersMap;
+
+  public CustomRequest(Request request, Map<String,FormParameter> formParameters){
+    this.request = request;
```

Comment:
ImmutableRequest does not support formParameters. Other Requests do not allow adding the formParameters after creation.